src/optimizer/filter_combiner.cpp

```diff
@@ -156,11 +157,113 @@ bool FilterCombiner::HasFilters() {
+				if (comp_in_exp.children[i]->type != ExpressionType::VALUE_CONSTANT) {
+					//! This indicates the column has a comparison that is not with a constant
+					not_constants.insert(column_ids[column_ref.binding.column_index]);
+					break;
+				} else {
+					auto &const_value_expr = (BoundConstantExpression &)*comp_in_exp.children[i].get();
+					auto it = checks.find(column_ids[column_ref.binding.column_index]);
+					if (it == checks.end()) {
+						checks[column_ids[column_ref.binding.column_index]] = {const_value_expr.value,
+						                                                       const_value_expr.value};
+					} else {
+						if (it->second.first > const_value_expr.value) {
+							it->second.first = const_value_expr.value;
+						}
+						if (it->second.second < const_value_expr.value) {
+							it->second.second = const_value_expr.value;
+						}
+					}
+				}
+			}
+		}
+		break;
+	}
+	case ExpressionType::COMPARE_EQUAL:
+	case ExpressionType::COMPARE_LESSTHAN:
+	case ExpressionType::COMPARE_LESSTHANOREQUALTO:
+	case ExpressionType::COMPARE_GREATERTHANOREQUALTO:
+	case ExpressionType::COMPARE_GREATERTHAN: {
+		auto &comp_exp = (BoundComparisonExpression &)*filter;
+		if ((comp_exp.left->expression_class == ExpressionClass::BOUND_COLUMN_REF &&
+		     comp_exp.right->expression_class == ExpressionClass::BOUND_CONSTANT)) {
+			auto &column_ref = (BoundColumnRefExpression &)*comp_exp.left;
+			auto &constant_value_expr = (BoundConstantExpression &)*comp_exp.right;
+			auto it = checks.find(column_ids[column_ref.binding.column_index]);
+			if (it == checks.end()) {
+				checks[column_ids[column_ref.binding.column_index]] = {constant_value_expr.value,
+				                                                       constant_value_expr.value};
+			} else {
+				if (it->second.first > constant_value_expr.value) {
+					it->second.first = constant_value_expr.value;
+				}
+				if (it->second.second < constant_value_expr.value) {
+					it->second.second = constant_value_expr.value;
+				}
+			}
+		} else if (comp_exp.left->expression_class == ExpressionClass::BOUND_COLUMN_REF) {
+			//! This indicates the column has a comparison that is not with a constant
+			auto &column_ref = (BoundColumnRefExpression &)*comp_exp.left;
+			not_constants.insert(column_ids[column_ref.binding.column_index]);
+		}
+		break;
+	}
+	default:
+		break;
+	}
+}
+
+vector<TableFilter> FilterCombiner::GenerateZonemapChecks(vector<idx_t> &column_ids,
+                                                          vector<TableFilter> &pushed_filters) {
+	vector<TableFilter> zonemap_checks;
+	unordered_map<idx_t, std::pair<Value, Value>> checks;
+	unordered_set<idx_t> not_constants;
+	//! We go through the remaining filters and capture their min max
+	for (auto &filter : remaining_filters) {
+		FindZonemapChecks(column_ids, checks, not_constants, filter.get());
+	}
+	//! We construct the equivalent filters
+	for (auto not_constant : not_constants) {
```

Comment:
Could you expand on why you remove zone map filters when columns have comparisons with non-constant filters? It should not influence zone maps if there are other comparisons, no? e.g. if I have the condition:

```sql
x > 4 AND x < 8 AND x = y
```

I can still check x for `[4, 8]` in the zonemap, no?